tutorials/misc/plot_ecog.py

```diff
@@ -9,71 +11,114 @@
+# We then compute the signal power in certain frequency bands (e.g. 30-90 Hz).
+# We compute the power in gamma and alpha bands.
+gamma_power = np.sum(raw.copy().filter(30, 90).get_data() ** 2, axis=1)
+alpha_power = np.sum(raw.copy().filter(8, 12).get_data() ** 2, axis=1)
 
 ###############################################################################
 # We can then plot the locations of our electrodes on our subject's brain.
 #
 # .. note:: These are not real electrodes for this subject, so they
 #           do not align to the cortical surface perfectly.
 
-subjects_dir = mne.datasets.sample.data_path() + '/subjects'
-fig = plot_alignment(info, subject='sample', subjects_dir=subjects_dir,
+subjects_dir = sample_path + '/subjects'
+fig = plot_alignment(raw.info, subject='sample', subjects_dir=subjects_dir,
                      surfaces=['pial'])
 mne.viz.set_3d_view(fig, 200, 70)
 
 ###############################################################################
 # Sometimes it is useful to make a scatterplot for the current figure view.
 # This is best accomplished with matplotlib. We can capture an image of the
 # current mayavi view, along with the xy position of each electrode, with the
-# `snapshot_brain_montage` function.
+# `snapshot_brain_montage` function. We can then visualize for example the
+# gamma power on the brain using MNE and matplotlib functions.
 
 # We'll once again plot the surface, then take a snapshot.
-fig_scatter = plot_alignment(info, subject='sample', subjects_dir=subjects_dir,
-                             surfaces='pial')
+fig_scatter = plot_alignment(raw.info, subject='sample',
+                             subjects_dir=subjects_dir, surfaces='pial')
 mne.viz.set_3d_view(fig_scatter, 200, 70)
 xy, im = snapshot_brain_montage(fig_scatter, montage)
 
 # Convert from a dictionary to array to plot
-xy_pts = np.vstack([xy[ch] for ch in info['ch_names']])
+xy_pts = np.vstack([xy[ch] for ch in raw.info['ch_names']])
+
+# colormap to view spectral power
+cmap = 'viridis'
 
-# Define an arbitrary "activity" pattern for viz
-activity = np.linspace(100, 200, xy_pts.shape[0])
+# show power at higher frequencies
+vmin, vmax = np.percentile(gamma_power, [10, 90])
 
-# This allows us to use matplotlib to create arbitrary 2d scatterplots
-_, ax = plt.subplots(figsize=(10, 10))
+fig, ax = plt.subplots(figsize=(10, 10))
 ax.imshow(im)
-ax.scatter(*xy_pts.T, c=activity, s=200, cmap='coolwarm')
 ax.set_axis_off()
+sc = ax.scatter(*xy_pts.T, c=gamma_power, s=200,
+                cmap=cmap, vmin=vmin, vmax=vmax)
+ax.set_title("Gamma frequency (30-90 Hz)")
+fig.colorbar(sc, ax=ax)
+
+# show power between low frequency
+vmin, vmax = np.percentile(alpha_power, [10, 90])
+
+fig, ax = plt.subplots(figsize=(10, 10))
+ax.imshow(im)
+ax.set_axis_off()
+sc = ax.scatter(*xy_pts.T, c=alpha_power, s=200,
+                cmap=cmap, vmin=vmin, vmax=vmax)
+ax.set_title("Alpha frequency (8-12 Hz)")
+fig.colorbar(sc, ax=ax)
```

Comment:
Seems like this could be more DRY... something like:
```python
fig, axs = plt.subplots(1, 2, figsize=(20, 10))
vmin, vmax = 'determined_jointly_for_both_bands'
for ax, band_power, band in zip(axs, [gamma_power, alpha_power], ['Gamma', 'Alpha']):
    ax.imshow(im)
    ax.set_axis_off()
    sc = ax.scatter(*xy_pts.T, c=band_power, s=200, cmap=cmap, vmin=vmin, vmax=vmax)
    ax.set_title(f'{band} frequency')
fig.colorbar(sc, ax=ax)
```